node/src/daser.rs

```diff
@@ -548,32 +592,65 @@ mod tests {
         sleep(Duration::from_millis(10)).await;
 
         // Sample block 10
-        handle_get_shwap_cid(&mut handle, &store, 10, &edses[9], false).await;
+        handle_get_shwap_cid(&mut handle, &store, &mut event_sub, 10, &edses[9], false).await;
 
         // Sample block 20
-        handle_get_shwap_cid(&mut handle, &store, 20, &edses[19], false).await;
+        handle_get_shwap_cid(&mut handle, &store, &mut event_sub, 20, &edses[19], false).await;
 
         // Sample and reject block 19
-        handle_get_shwap_cid(&mut handle, &store, 19, &edses[18], true).await;
+        handle_get_shwap_cid(&mut handle, &store, &mut event_sub, 19, &edses[18], true).await;
 
         // Simulate disconnection
         handle.announce_all_peers_disconnected();
+
+        // Daser may scheduled Block 18 already, so we need to reply to that requests.
+        // For the sake of the test we reply with a bitswap timeout.
```

Comment:
actually I think the bitswap timeout is what would happen anyway if all peers disconnected :+1: